Simplified/Reader2/UI/NYPLBaseReaderViewController.swift

```diff
@@ -261,7 +263,63 @@ class NYPLBaseReaderViewController: UIViewController, Loggable {
       updateBookmarkButton(withState: false)
     }
   }
+  
+  //----------------------------------------------------------------------------
+  // MARK: - Read Position
+  
+  private func storeReadPosition(locator: Locator) {
+    // Avoid overwriting location when reader first open
+    guard (locator.locations.totalProgression ?? 0) != 0 else {
+      return
+    }
+    let bookLocation = NYPLBookLocation(locator: locator, publication: publication, renderer: NYPLBookLocation.r2Renderer)
+    NYPLBookRegistry.shared().setLocation(bookLocation, forIdentifier: book.identifier)
+    bookmarksBusinessLogic.postReadPosition(locationString: bookLocation.locationString)
+  }
 
+  /// Restore locally stored location and navigate to it
+  /// Then sync location from server and present alert for navigation
+  private func restoreReadPosition() {
```

Comment:
most of this is also business logic, but we could refactor this in a future ticket if that's more practical.

Reply:
Refactored to bookmark business logic